### PR TITLE
Minor improvement to Vulkan pipeline state and bindings management

### PR DESCRIPTION
**Pull request by gdkchan**
This improves the management of pipeline state and bindings management on the Vulkan backend:
- #3793 started using dynamic state for blend constants, which reduces  the amount of pipelines that we need to create, however the state was still there, and was still being compared/hashed which is a bit wasteful. This PR removes the blend constant from the pipeline state since it is not used anymore. Should make pipeline state lookup a little bit faster.
- Changed the descriptor set updater to stop ignoring null textures and clean up the bindings, instead of keeping stale bindings. This is to avoid bugs where we keep potentially already freed resources bounds, and also the potential for "garbage" to be sampled on the shader. The new behaviour should be closer to the hardware behaviour, and OpenGL was changed too.
- Avoid one array allocation/copy on the ASTC redundant update check path. This is something that I noticed a while ago and was pointed out by one of our contributors recently. Decided to do the change here because I felt like it was not necessary to open a separate PR for such a minor change.

I wanted to remove `BlendConstant` from the `BlendDescriptor` too. The `BlendDescriptor` is passed for each blend, but this is not really necessary because there's only one blend constant shared by all blends. However since this is currently stored in the shader cache, changing the struct would break compatibility with existing shader caches so I just kept it as-is.

Testing is welcome to make sure nothing broke, but I think chances are pretty low with this one.

### CODE REVIEW

File: Ryujinx.Graphics.OpenGL/Image/TextureBase.cs [hunk 1]
```diff
@@ -34,5 +34,22 @@ protected void Bind(TextureTarget target, int unit)
             GL.ActiveTexture(TextureUnit.Texture0 + unit);
             GL.BindTexture(target, Handle);
         }
+
+        public static void ClearBinding(int unit)
+        {
+            GL.ActiveTexture(TextureUnit.Texture0 + unit);
+
+            // Clear all possible targets since we don't know which one the shader will access.
+            GL.BindTexture(TextureTarget.Texture1D, 0);
```
Comment (riperiperi) [inline Ryujinx.Graphics.OpenGL/Image/TextureBase.cs:43]: What about this?
https://registry.khronos.org/OpenGL-Refpages/gl4/html/glBindTextureUnit.xhtml

File: Ryujinx.Graphics.Gpu/Image/Texture.cs [hunk 2]
```diff
@@ -708,11 +708,12 @@ public void SynchronizeFull()
                 else
                 {
                     bool dataMatches = _currentData != null && data.SequenceEqual(_currentData);
-                    _currentData = data.ToArray();
                     if (dataMatches)
                     {
                         return;
                     }
+
+                    _currentData = data.ToArray();
```
Comment (riperiperi) [inline Ryujinx.Graphics.Gpu/Image/Texture.cs:716]: Ouch, nice catch here.

**Comment by Thorbenl** [comment]
FYI: this fixed lots of issue sofr Pokemon Scarlet and Violet :)

File: Ryujinx.Graphics.OpenGL/Image/TextureBase.cs (hunk 1 above)
Comment (gdkchan) [reply Ryujinx.Graphics.OpenGL/Image/TextureBase.cs]: Nice, I didn't know this was a thing, updated.

**Comment by Frizzacazzi** [comment]
> AMD still does

Experiencing the same.

**Comment by riperiperi** [comment]
Experiencing the same what?

**Comment by Frizzacazzi** [comment]
> Experiencing the same what?

AMD Radeon Pro 450 still yellowscreens. Sorry for not being clear before.

**Review by help-14 — APPROVED** [review]

**Review by marysaka — APPROVED** [review]

**Pull request merged**